cpp/ManagedOutputStream.cpp

```diff
@@ -34,6 +35,10 @@ class ManagedOutputStream final : public arrow::io::OutputStream
 
 	~ManagedOutputStream()
 	{
+		arrow::Status st = this->Close();
+		if (!st.ok()) {
+			ARROW_LOG(FATAL) << "Error ignored when destroying ManagedOutputStream: " << st;
```

Comment:
Should this be FATAL if it's ignored and we carry on regardless? ERROR perhaps?

Reply:
Yes it should be ERROR.